crates/octez/src/async/protocol.rs

```diff
@@ -183,6 +199,25 @@ impl ProtocolParameterBuilder {
         json.insert("bootstrap_accounts".to_owned(), Value::from(&accounts));
         Ok(())
     }
+
+    fn merge_bootstrap_contracts(
+        &mut self,
+        json: &mut serde_json::Map<String, Value>,
+    ) -> anyhow::Result<()> {
+        let mut contracts = BootstrapContracts::default();
+        if let Some(value) = json.get("bootstrap_contracts") {
+            if let Some(existing_contracts) = value.as_array() {
+                for contract in existing_contracts {
+                    contracts.add_contract(BootstrapContract::try_from(contract)?);
+                }
+            }
+        }
```

Comment:
If Deserializer is implemented for `BootstrapContracts`, then you could do `let bootstrap_contracts: BoostramContract = serde_json::from_value(value)` which is much cleaner.

Reply:
Updated